### PR TITLE
Fe feature add forgot password page

### PR DESCRIPTION
**Pull request by Sidragon123**
Frontend integration for the forgot password feature
Step 1
image
Step 2
image
 Step 3
image
Success
image

### CODE REVIEW

File: views/src/Pages/ForgotPassword.jsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      errorMsg.current.textContent =
+        "Password must be at least 8 characters long";
+    }
+
+    if (newPwd.length > 30) {
+      errorMsg.current.textContent =
+        "Password cannot be more than 30 characters long";
+    }
+  };
+  const nextStep = async () => {
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx:47]: This whole function has too many nested if else statement making it difficult to understand.  
I suggest either breaking out the various steps into their own functions and using them or refactoring so we
don't have too many if else nested. Look into guard clauses and use them where they apply.

File: views/src/Pages/ForgotPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      }
+    }
+  };
+
+  const prevStep = () => {
+    errorMsg.current.textContent = ``;
+    setActive((current) => (current > 0 ? current - 1 : current));
+  };
+  const finishUp = async () => {
+    console.log(newPwd, confirmPwd);
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx]: We don't want to have this console log here. Remove this

File: views/src/Pages/ForgotPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            setEmailSubmitted(true);
+          } else if (response.status === 404) {
+            errorMsg.current.textContent = `Email not registered!`;
+          }
+        } else {
+          errorMsg.current.textContent = `Error! Enter valid email format.`;
+        }
+      } else if (active === 1) {
+        // Make the API call to check Token endpoint
+        console.log(otp, email);
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx]: Remove this console log as well

File: views/src/Pages/ForgotPassword.jsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+import { baseUrl } from "../API/index";
+
+
+
+function ForgotPassword() {
+  const [active, setActive] = useState(0);
+  const [email, setEmail] = useState("");
+  const [emailSubmitted, setEmailSubmitted] = useState(false);
+  const [otpValidated, setOtpValidated] = useState(false);
+  const errorMsg = useRef(null);
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx]: We don't need to use useRef here. useState hook is a better option for this

File: views/src/Pages/ForgotPassword.jsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  const [otp, setOtp] = useState(0);
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx]:
```suggestion
    const [otp, setOtp] = useState("");
```

File: views/src/Pages/ForgotPassword.jsx [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+  const [newPwd, setNewPwd] = useState(null);
+  const [confirmPwd, setConfirmPwd] = useState(null);
+  const updatepasswordValidator = (input, confirmation) => {
```
Comment (gbudjeakp) [inline views/src/Pages/ForgotPassword.jsx:26]: This function checks but only sets the error message for the first failing condition. I'd look into accumulating the error.

**Review by gbudjeakp — CHANGES_REQUESTED** [review]

File: views/src/Pages/ForgotPassword.jsx (hunk 1 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx:47]: Implemented simpler syntax so the max depth is 2 nested statements

File: views/src/Pages/ForgotPassword.jsx (hunk 2 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx]: Removed

File: views/src/Pages/ForgotPassword.jsx (hunk 4 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx]: Switched error display infrastructure to a state array as opposed to directly writing into the DOM

File: views/src/Pages/ForgotPassword.jsx (hunk 5 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx]: Resolved

File: views/src/Pages/ForgotPassword.jsx (hunk 6 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx:26]: Applied cumulative error checking and multiple error display using state arrays

File: views/src/Pages/ForgotPassword.jsx (hunk 3 above)
Comment (Sidragon123) [reply views/src/Pages/ForgotPassword.jsx]: resolved

**Review by gbudjeakp — APPROVED** [review]

**Pull request merged**